map_view/display-device-location-with-autopan-modes/src/main/java/com/esri/samples/display_device_location_with_autopan_modes/DisplayDeviceLocationWithAutopanModesSample.java

```diff
@@ -0,0 +1,200 @@
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.LocationDisplay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+import org.apache.commons.io.IOUtils;
+
+public class DisplayDeviceLocationWithAutopanModesSample extends Application {
+
+  private MapView mapView;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create the stack pane and the application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set a title, size, and add the scene to stage
+      stage.setTitle("Display Device Location With Autopan Modes Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+      scene.getStylesheets().add(getClass().getResource("/display_device_location_with_autopan_modes/style.css").toExternalForm());
+
+      // create an ArcGISMap with the imagery basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+
+      // create a view and add the ArcGISMap to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create a combo box
+      ComboBox<String> comboBox = new ComboBox<>();
+      comboBox.setMaxWidth(Double.MAX_VALUE);
+      comboBox.setDisable(true);
+      // add the autopan modes to the combo box
+      comboBox.getItems().addAll("Recenter", "Navigation", "Compass", "Off");
+      comboBox.setValue("Recenter");
+
+      // add a label
+      Label autopanModeLabel = new Label("Choose an autopan mode:");
+      // add a checkbox that toggles the visibility of the location display
+      CheckBox checkBox = new CheckBox("Show device location");
+      checkBox.setDisable(true);
+
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream(
+        "/display_device_location_with_autopan_modes/polyline_data.json"), StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline locations = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWgs84());
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        locations, new SimulationParameters(Calendar.getInstance(), 5.0, 0.0, 0.0));
+
+      // configure the map view's location display to follow the simulated location data source
+      LocationDisplay locationDisplay = mapView.getLocationDisplay();
+      locationDisplay.setLocationDataSource(simulatedLocationDataSource);
+      locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      locationDisplay.setInitialZoomScale(1000);
+
+      // toggle the location display visibility on check
```

Comment:
these comments aren't entirely accurate as to what the API is doing. 

Whilst `startAsync()` and `stop()` do indeed show/hide the location display, what it's doing at an API level is initializing and then receiving location data source updates. When stopping, it ceases to receive the location updates. 

I'd therefore suggest the following: 

```suggestion
      // control location display updates and visibility
```